test/cbmc/stubs/agent_command_functions_stub.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * coreMQTT-Agent v1.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file agent_command_functions_stub.c
+ * @brief Function stubs for processing an MQTT agent command.
+ */
+
+/* MQTT Agent command functions include. */
+#include "mqtt_agent_command_functions.h"
+
+static MQTTStatus_t MQTTAgentCommand_Stub( MQTTAgentContext_t * pMqttAgentContext,
+                                           void * pArg,
+                                           MQTTAgentCommandFuncReturns_t * pReturnFlags )
+{
+    MQTTStatus_t returnStatus;
+    uint16_t packetId;
+
+    __CPROVER_assert( pMqttAgentContext != NULL, "MQTT Agent context is not NULL." );
+    __CPROVER_assert( pReturnFlags != NULL, "Return flags pointer is not NULL." );
+
+    __CPROVER_assume( packetId > 0U );
+
+    pReturnFlags->packetId = packetId;
+    pReturnFlags->endLoop = nondet_bool();
+    pReturnFlags->addAcknowledgment = nondet_bool();
+    pReturnFlags->runProcessLoop = nondet_bool();
```

Comment:
Would it be better to just set it all to `false` by default since the actual functions would just memset the entire struct to 0?

Reply:
I think it is better to be independent to the maximum extend of how we have implemented the command functions in testing memory safety. This will give us a good confidence that any future changes in these implementations are also memory safe.